pkg/ssh/auth.go

```diff
@@ -95,9 +95,14 @@ func (a *Auth) handlePublicKeyMethodRequest(
 ) (PublicKeyAuthMethodResponse, error) {
 	sessionID, err := a.resolveSessionIDFromFingerprint(ctx, req.PublicKeyFingerprintSha256)
 	if err != nil {
-		if !databroker.IsNotFound(err) {
-			return PublicKeyAuthMethodResponse{}, err
+		st, ok := status.FromError(err)
+		if ok && (st.Code() == codes.NotFound || st.Code() == codes.InvalidArgument) {
```

Comment:
InvalidArgument is too likely to cause regressions here IMO (since it is much more common), we should stick with NotFound only and change the expiration error to be NotFound.